requires.py

```diff
@@ -106,7 +106,9 @@ def get_auth_credentials(self, user):
         """
         Return the authentication credentials.
         """
-        rx = self.all_joined_units.received.get('creds')
+        rx = {}
+        for unit in self.all_joined_units:
+            rx.update(unit.received.get('creds'))
```

Comment:
If `unit.received.get('creds')` returns `None`, an exception will occur here. I recommend this instead:

```
rx.update(unit.received.get('creds', {}))
```

Reply:
Thanks for the review